images/dev.Dockerfile

```diff
@@ -140,9 +140,15 @@ RUN if [ "$TARGETARCH" = "amd64" ]; then \
         rm -f minio.rpm; \
     fi
 
+# Install pre-commit
+RUN yum install -y python39 &&\
+    /bin/pip-3.9 install pre-commit
+
+COPY ./images/dev-container-entrypoint.sh /usr/local/bin/
+
 WORKDIR /app
 
-CMD ["/bin/bash"]
+ENTRYPOINT ["dev-container-entrypoint.sh"]
```

Comment:
Why not just do the `pre-commit install` in the Dockerfile and skip having a `CMD` or `ENTRYPOINT`?

Reply:
If you mean by `RUN pre-commit install` at the end of the Dockerfile, then the answer would be No, because `pre-commit install` installs hooks to `.git` folder, which needs to be ran after the container binds the project folder and I think the only way is to add to the entrypoint script so that the command runs when the container has started.

Reply:
Ahh, makes sense. Thanks for the clarification.